src/wasm-lib/kcl/src/std/patterns.rs

```diff
@@ -70,6 +74,237 @@ impl LinearPattern {
     }
 }
 
+/// A linear pattern
+/// Each element in the pattern repeats a particular piece of geometry.
+/// The repetitions can be transformed by the `transform` parameter.
+pub async fn pattern_transform(args: Args) -> Result<MemoryItem, KclError> {
+    let (num_repetitions, transform, extr) = args.get_pattern_transform_args()?;
+
+    let extrude_groups = inner_pattern_transform(
+        num_repetitions,
+        FunctionParam {
+            inner: transform.func,
+            fn_expr: transform.expr,
+            meta: vec![args.source_range.into()],
+            ctx: args.ctx.clone(),
+            memory: args.current_program_memory.clone(),
+        },
+        extr,
+        &args,
+    )
+    .await?;
+    Ok(MemoryItem::ExtrudeGroups { value: extrude_groups })
+}
+
+/// A linear pattern on a 3D solid.
+/// Each repetition of the pattern can be transformed (e.g. scaled, translated, hidden, etc).
+///
+/// ```no_run
+/// // Parameters
+/// const r = 50    // base radius
+/// const h = 10    // layer height
+/// const t = 0.005 // taper factor [0-1)
+/// // Defines how to modify each layer of the vase.
+/// // Each replica is shifted up the Z axis, and has a smoothly-varying radius
+/// fn transform = (replicaId) => {
+///   let scale = r * abs(1 - (t * replicaId)) * (5 + cos(replicaId / 8))
+///   return {
+///     translate: [0, 0, replicaId * 10],
+///     scale: [scale, scale, 0],
+///   }
+/// }
+/// // Each layer is just a pretty thin cylinder with a fillet.
+/// fn layer = () => {
+///   return startSketchOn("XY") // or some other plane idk
+///     |> circle([0, 0], 1, %, 'tag1')
+///     |> extrude(h, %)
+///     // |> fillet({
```

Comment:
since these go on the website id either delete this or uncocmment it

Reply:
ah good call